controllers/story.js

```diff
@@ -1,431 +1,449 @@
-    MongoClient.connect(secrets.db, function(err, database) {
-        database.collection('stories').find({
-            "$text": {
-                "$search": req.body.data.searchValue
-            }
-        }, {
-            headline: 1,
-            timePosted: 1,
-            link: 1,
-            description: 1,
-            rank: 1,
-            upVotes: 1,
-            author: 1,
-            comments: 1,
-            image: 1,
-            storyLink: 1,
-            metaDescription: 1,
-            textScore: {
-                $meta: "textScore"
-            }
-        }, {
-            sort: {
-                textScore: {
-                    $meta: "textScore"
-                }
-            }
-        }).toArray(function(err, items) {
-            if (items !== null && items.length !== 0) {
-                return res.json(items);
-            }
-            return res.status(404);
-        });
+  MongoClient.connect(secrets.db, function(err, database) {
+    if (err) {
+      return res.status(500);
+    }
+    database.collection('stories').find({
+      '$text': {
+        '$search': req.body.data.searchValue
+      }
+    }, {
+      headline: 1,
+      timePosted: 1,
+      link: 1,
+      description: 1,
+      rank: 1,
+      upVotes: 1,
+      author: 1,
+      comments: 1,
+      image: 1,
+      storyLink: 1,
+      metaDescription: 1,
+      textScore: {
+        $meta: 'textScore'
+      }
+    }, {
+      sort: {
+        textScore: {
+          $meta: 'textScore'
+        }
+      }
+    }).toArray(function(err, items) {
+      if (err) {
+        return res.status(500);
+      }
+      if (items !== null && items.length !== 0) {
+        return res.json(items);
+      }
+      return res.status(404);
     });
+  });
 };
 
 exports.upvote = function(req, res, next) {
-    var data = req.body.data;
-    Story.find({'_id': data.id}, function(err, story) {
-        if (err) {
-            res.status(500);
-            return next(err);
-        }
-        story = story.pop();
-        story.rank++;
-        story.upVotes.push(
-            {
-                upVotedBy: data.upVoter._id,
-                upVotedByUsername: data.upVoter.profile.username
-            }
-        );
-        story.markModified('rank');
-        story.save();
-        return res.send(story);
+  var data = req.body.data;
+  Story.find({'_id': data.id}, function(err, story) {
+    if (err) {
+      res.status(500);
+      return next(err);
+    }
+    story = story.pop();
+    story.rank++;
+    story.upVotes.push(
+      {
+        upVotedBy: data.upVoter._id,
+        upVotedByUsername: data.upVoter.profile.username
+      }
+    );
+    story.markModified('rank');
+    story.save();
+    User.find({'_id': story.author.userId}, function(err, user) {
+      'use strict';
+      if (err) {
+        return next(err);
+      }
+      user = user.pop();
+      user.progressTimestamps.push(Date.now());
+      user.save();
     });
```

Comment:
There are a couple of async calls here that need to be nested or promisify and chained here.